pkg/apis/core/validation/validation.go

```diff
@@ -5357,6 +5362,11 @@ func ValidateResourceRequirements(requirements *core.ResourceRequirements, fldPa
 		}
 		if helper.IsHugePageResourceName(resourceName) {
 			reqContainsHugePages = true
+			if !helper.IsHugePageResourceValueValid(resourceName, quantity) {
+				reqContainsHugePages = false
+				allErrs = append(allErrs, field.Invalid(fldPath, quantity.String(),
+					fmt.Sprintf("value must be integer multiple of %s limit", resourceName)))
```

Comment:
same here

Reply:
nit: I'd suggest to reorganize this code a little for better readability, e.g.:
```
if helper.IsHugePageResourceValueValid(resourceName, quantity) {
    reqContainsHugePages = True
} else {
    allErrs = append(allErrs, ...
}
```

Reply:
Another idea is to move this check into ValidateResourceQuantityValue as essentially it is a resource quantity check. wdyt?

Reply:
ValidateResourceQuantityValue is used for check various kinds of resource quantity value, such as node.Status.Capacity. Instead, I add a new function validateResourceQuantityHugePageValue so that the impact of the verification is smaller and IsHugePageResourceValueValid() is only called once here.